django/db/models/sql/query.py

```diff
@@ -2385,7 +2380,17 @@ def set_values(self, fields):
                         extra_names.append(f)
                     elif f in self.annotation_select:
                         annotation_names.append(f)
+                    elif f in self.annotations:
+                        raise FieldError(
+                            f"Cannot select the '{f}' alias. Use annotate() to "
+                            "promote it."
+                        )
                     else:
+                        # Call `names_to_path` to ensure a FieldError including
+                        # annotations about to be masked as valid choices if
+                        # `f` is not resolvable.
+                        if self.annotation_select:
```

Comment:
This branch is not strictly necessary but it's gates the somewhat expensive `names_to_path` function call that is unnecessary if not annotations exist.